### PR TITLE
Quiz back end

### PR DESCRIPTION
**Pull request by wintleh**
This PR adds the back end for the quiz page. This includes SQL commands to setup the database.

### CODE REVIEW

File: setup-db.sql [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+insert into Quiz_Answer values (9,  1, "definition4");
+insert into Quiz_Answer values (10, 1, "definition5");
+insert into Quiz_Answer values (11, 1, "definition6");
+
+insert into Quiz_Answer values (12, 2, "place1");
+insert into Quiz_Answer values (13, 2, "place2");
+insert into Quiz_Answer values (14, 2, "place3");
+insert into Quiz_Answer values (15, 2, "place4");
+insert into Quiz_Answer values (16, 2, "place5");
+insert into Quiz_Answer values (17, 2, "place6");
```
Comment (wintleh) [inline setup-db.sql]: Questions and answers need to be filled in once we have all the data.

File: src/db/config.php [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+define('DBHOST', 'localhost');
+define('DBNAME', 'ekg_viewer');
+define('DBUSER', 'EKGViewer');
+define('DBPASS', 'wit123');
```
Comment (wintleh) [inline src/db/config.php:5]: Uses EKGViewer user to access the DB

File: src/db/database.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?php
+require_once("config.php");
+
+class Database {
+
+    function __construct() { }
+
+    function selectQuery($query) 
+    {
```
Comment (wintleh) [inline src/db/database.php:9]: Function handles the DB connection and returns an array of data for select commands

File: src/db/quiz.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+    function getInterpretQuestion($questionNumber)
+    {
+        return $this->getQuestion($questionNumber, "how-to-interpret", "img");
+    }
+
+    function getPlaceQuestion($questionNumber)
+    {
+        return $this->getQuestion($questionNumber, "how-to-place", "img");
+    }
```
Comment (wintleh) [inline src/db/quiz.php]: Handles all the DB queries and creates the question and answer objects which are used to generate html.

File: src/quiz.php [hunk 5]
```diff
@@ -19,93 +20,34 @@
...
-                )
-            ))->output();
-            ?>
-            <input type="submit" value="Submit" class="form-button">
-            <input type="reset" value="Clear" class="form-button">
-            <h2 id="submission-error-message"></h2>
-        </form>
+        <?php
+        $quiz = new Quiz();
+
```
Comment (wintleh) [inline src/quiz.php:25]: Uses the Quiz class which handles all the DB queries

File: src/quiz.php [hunk 6]
```diff
@@ -19,93 +20,34 @@
...
+        $placeQuestion      = $quiz->getPlaceQuestion(3);
+        $interpretQuestion  = $quiz->getInterpretQuestion(4);
+
+        // Ensure that all questions were generated successfully
+        if (
+            $historyQuestion    != null &&
+            $definitionQuestion != null &&
+            $placeQuestion      != null &&
+            $interpretQuestion  != null
+        ) {
```
Comment (wintleh) [inline src/quiz.php:37]: Basic error checking

File: src/results.php [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+<body>
+    <?php include "./templates/page-header.php" ?>
+    <section id="content-wrapper">
+        <?php
+            foreach ($_POST as $key => $value)
+            {
+                echo $key . ": " . $value . "<br/>";
+            }
+        ?>
```
Comment (wintleh) [inline src/results.php]: Just prints the post data. The keys are question ids and the values are the chosen answer ids. These ids are the ones from the DB, so a query using that id can get the rest of the question/answers data.

File: README.md [hunk 8]
```diff
@@ -5,39 +5,52 @@ An online platform for learning about EKG.
...
 
-End with an example of getting some data out of the system or using it for a little demo
+- On the top of the page, select the "Structure" tab, and verify that there are three tables: quiz_answer, quiz_category, and quiz_question.
+
+#### 3. Verify installation
+- Finally, navigate to ```http://localhost/EKGViewer/src/quiz.php``` in your browser. 
+  
+- If there are questions and answers appearing on the page, then everything is setup correctly. Refresh the page to generate new questions. 
+  
+- If there are any errors return to steps 1 and 2 and verify that the repo was cloned to the correct directory and the database was setup correctly. If there are still issues after that, verify your XAMPP installation.
```
Comment (wintleh) [inline README.md]: Added setup steps to the README

**Review by wintleh — COMMENTED** [review]
The important things to note are the POST data and the DB user/pass.

**Comment by am2021** [comment]
image
not sure if i have this setup for review

**Review by roched1atwit — APPROVED** [review]
Starting the Results Page rn, I'll let you know if  i need help getting the results from their quiz

**Pull request merged**